stdlib/public/TensorFlow/CompilerRuntime.swift

```diff
@@ -394,6 +579,154 @@ public final class _ExecutionContext {
   }
 }
 
+// Elements in `outputs` can come from two sources:
+// a) Symbolic tensors produced by tensor ops, and added as trace graph nodes.
+// b) Concrete tensors produced by host code (e.g. Tensor(1.0)).
+func finalizeAndExecuteTraceFn(fnName: String,
+                               inputs: [Tensor<Float>],
+                               outputs: [CTensorHandle]) -> [CTensorHandle] {
+  guard let tracedFn = _RuntimeConfig.traceState.getFunction else {
+    fatalError("Not in tracing mode!.")
+  }
+  _RuntimeConfig.traceState = .notTracing
+
+  return tracedFn.finalizeAndExecute(traceeBasicName: fnName,
+                                     inputs: inputs,
+                                     outputs: outputs)
+}
+
+public extension TensorArrayProtocol {
```

Comment:
This is not used outside this module. Make private.
```suggestion
private extension TensorArrayProtocol {
```

Reply:
Done